File: 1017_TLE.rb
n = gets.to_i
f = Array.new(n+1) { Array.new(n+1) { 0 } }

f[0][0] = 1
0.upto(n-1) do |i|
	0.upto(i) do |j|
		(j+1).upto(n-i) do |k|
			f[i+k][k] += f[i][j]
		end
	end
end

ans = -1
f[n].each { |x| ans += x }
print ans, ?\n

File: 1017_TLE.py
n = int(raw_input())
f = [[0 for i in range(n+1)]for j in range(n+1)]

f[0][0] = 1
for i in range(n) :
	for j in range(i+1) :
		for k in range(j+1, n-i+1) :
			f[i+k][k] += f[i][j]

ans = 0
for i in range(n) :
	ans += f[n][i]

print ans
